### ISSUE TITLE
There is no navigation buttons in toolbar.

### ISSUE THREAD

**Issue by chansey97** [pkulchenko/ZeroBraneStudio#690]
In VisualStudio/IntelliJ IDEA/Eclipse, we have navigation buttons in toolbar, it's very convenience when browsing code.
I think it is necessary to add this feature.
Thanks.

**Comment by pkulchenko** [comment]
Do you have an screenshot/description of this feature?

**Comment by chansey97** [comment]
![untitled](https://cloud.githubusercontent.com/assets/4015056/18017317/78856e32-6c03-11e6-9efa-9f50547fb87d.png)
@pkulchenko I captured a screenshot. You can this two buttons to navigate code.

**Comment by pkulchenko** [comment]
So, that would jump between cursor locations you visited? I'm not familiar with this feature...

**Comment by chansey97** [comment]
Yes,  it's very convenience when browsing code.

**Comment by pkulchenko** [comment]
It's still unclear to me how you use this navigation. Could you provide a scenario?

**Comment by chansey97** [comment]
This feature is important when you reading other people's code.
For example:
There are 3 modules(A B and C) in your project. .
The programm's main flow is "module A call module B, module B call module C".
When you reading this code, you may read the function in module A first, then read the function in module B and C. After fininsh reading the function in module C, you can press Navigate Backward to module A. (back to main flow again)

**Comment by halfnelson** [comment]
Also good for navigation within the same file, I can right click on a function call and "Goto Definition", then read the definition of the function, then hit the back button to go back to the function call that I navigated from

**Comment by pkulchenko** [comment]
@halfnelson, I agree; in fact we've already [discussed it in the past](http://www.freelists.org/post/zerobrane/Request-ForwardBackwards-navigation). The trouble is to figure out what to add to the queue and how to navigate it.

`Go to Definition` is covered: you can use `Alt-Left` to jump to a previous location. I should probably add a message to the status bar that you can jump back. This only works after couple of shortcuts and can probably be extended to cover more cases.

**Comment by robertlzj** [comment]
> [@halfnelson](https://github.com/halfnelson), I agree; in fact we've already [discussed it in the past](http://www.freelists.org/post/zerobrane/Request-ForwardBackwards-navigation). The trouble is to figure out what to add to the queue and how to navigate it.

Hi, some scenario is, when project has more than one files and many tabs open, after jump to another file/tab, maybe from project tree to check something, then want to jump back.
Currently, didn't remember which tab we start from exactly, and need to find it among tabs, so, backward navigtion is usefull.

So, I think queue navigation among any caret movement by click, between files?

**Comment by pkulchenko** [comment]
> Currently, didn't remember which tab we start from exactly, and need to find it among tabs, so, backward navigtion is usefull.

@robertlzj, is this not what `Ctrl-,` does (from the `File | Recent Files` menu)?

**Comment by robertlzj** [comment]
> is this not what `Ctrl-,` does (from the `File | Recent Files` menu)?

😄 I was thought it is used only to restore recent closed tab.
So the navigation forward/backward on files could use `Ctrl-,`, `Ctrl-.`.

**Comment by pkulchenko** [comment]
> I was thought it is used only to restore recent closed tab.

Yes, it will do that as well, but it also works for tabs that are not closed ;).

> So the navigation forward/backward on files could use Ctrl-,, Ctrl-..

Correct. You can also change it to a different mapping if you want, but the mnemonic rule is that it's `<` key to go back and `>` key to go forward.
